ods_ci/utils/scripts/fetch_tests.py

```diff
@@ -0,0 +1,164 @@
+import argparse
+import os
+
+from robot.model import SuiteVisitor
+from robot.running import TestSuiteBuilder
+from util import execute_command
+
+
+class TestCasesFinder(SuiteVisitor):
+    def __init__(self):
+        self.tests = []
+
+    def visit_test(self, test):
+        self.tests.append(test)
+
+
+def get_repository(test_repo):
+    """
+    If $test_repo is a remote repo, the function clones it in "ods-ci-temp" directory.
+    If $test_repo is a local path, the function checks the path exists.
+    """
+    repo_local_path = "./ods_ci/ods-ci-temp"
+    if "http" in test_repo or "git@" in test_repo:
+        print("Cloning repo ", test_repo)
+        ret = execute_command(
+            "git clone {test_repo} {output_dir}".format(test_repo=test_repo, output_dir=repo_local_path)
+        )
+        if "error" in ret.lower():
+            # actual error gets printed during "execute_command"
+            raise Exception("Failed to clone the given repository")
+    elif not os.path.exists(test_repo):
+        raise FileNotFoundError("local path {} was not found".format(test_repo))
+    else:
+        print("Using local repo ", test_repo)
+        repo_local_path = test_repo
+    return repo_local_path
+
+
+def checkout_repository(ref):
+    """
+    Checkouts the repository at current directory to the given branch/commit ($ref)
+    """
+    ret = execute_command("git checkout {}".format(ref))
+    if "error" in ret.lower():
+        # actual error gets printed during "execute_command"
+        raise Exception("Failed to checkout to the given branch/commit {}".format(ref))
+    ret = execute_command("git checkout")
+    print(ret)
```

Comment:
I think the execute_command already print the return value so maybe we can remove this line

Reply:
done